### PR TITLE
[WEBSITE] Blog post about DataFusion 13.0.0

### PR DESCRIPTION
**Pull request by alamb**
re https://github.com/apache/arrow-datafusion/issues/3671

This blog describes what has been going on in DataFusion for the last 5 months

### CODE REVIEW

**Comment by alamb** [comment]
@andygrove  mentions there is a draft blog for datafusion 11 that was not published that we can use for additional content: https://docs.google.com/document/d/1tPCgeB6iQPVvbRyaXft7nKqorrhv-XDqVXuMp4pG9ns/edit?usp=sharing

File: _posts/2022-10-20-datafusion-13.0.0.md [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+
+One of DataFusion’s advantages is its implementation in [Rust](https://www.rust-lang.org/) and thus its easy integration with the broader Rust ecosystem. Rust continues to be a major source of benefit, from the [ease of parallelization with the high quality and standardized `async` ecosystem](https://www.influxdata.com/blog/using-rustlangs-async-tokio-runtime-for-cpu-bound-tasks/) , as well as its modern dependency management system and wonderful performance. <!-- I wonder if we should link to clickbench?? -->
```
Comment (isidentical) [inline _posts/2022-10-20-datafusion-13.0.0.md:48]: > I wonder if we should link to clickbench??

Did the clickbench results got updated with 13.0? AFAIK we should be much faster than we were compared to the initial integration time (there were a lot of slowness coming from SelectK queries and a few other optimizations like `regex_replace`, which we should handle much better now). CC: @waitingkuo

File: _posts/2022-10-20-datafusion-13.0.0.md [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+ - Custom / Dynamic table provider factories (#3311)
+ - `SHOW CREATE TABLE` for support for views (#2830)
+
+# Faster Execution
+ - Optimizations of TopK (queries with a `LIMIT` or `OFFSET` clause):  (#3527), (#2521)
+ - Reduce `left`/`right`/`full` joins to `inner` join (#2750)
+ - Convert  cross joins to inner joins when possible (#3482)
+ - Sort preserving `SortMergeJoin` (#2699)
+ - Improvements in group by and sort performance (#2375)
+
```
Comment (isidentical) [inline _posts/2022-10-20-datafusion-13.0.0.md:100]: Adaptive `regex_replace` implementation: https://github.com/apache/arrow-datafusion/issues/3518

File: _posts/2022-10-20-datafusion-13.0.0.md [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+# Parquet
+ - Parquet reader can now read directly from remote parquet files on object storage (#2489) (#2677) (#3051)
+ - Experimental support for “predicate pushdown” that implements late materialization after filtering during the scan (we plan another blog post on this soon).
+ - Support reading directly from S3 via `datafusion-cli ` (#3631)
+
+# DataType Support
+- Support for `IN` LIST,  support, casting, etc for `Decimal`
+- Timestamps: `date_bin` built-in function (#3034), timestamp plus minus interval (#3110), TIME literal values (#3010)
+- Binary operations (AND, XOR, etc):  (#3037) (#1619) (#3420) (#3430)
+
```
Comment (isidentical) [inline _posts/2022-10-20-datafusion-13.0.0.md:114]: - `IS TRUE/FALSE` and `IS [NOT] UNKNOWN` (#3235), (#3246)

File: _posts/2022-10-20-datafusion-13.0.0.md [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+
+
+## Upcoming Work
+With the community growing and code accelerating, there is so much great stuff on the horizon. Some features we expect to land in the next few months:
+
+- [Complete Parquet Pushdown](https://github.com/apache/arrow-datafusion/issues/3462)
+- [Additional date/time support](https://github.com/apache/arrow-datafusion/issues/3148)
+- improve / make it easier to implement FlightSQL using DataFusion (TODO LINK)
+
```
Comment (isidentical) [inline _posts/2022-10-20-datafusion-13.0.0.md:123]: Not sure if it's worth including, but maybe `Improved cost estimations & Nested Join Optimizations`. (#128, #3843, #3845)

File: _posts/2022-10-20-datafusion-13.0.0.md [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+    87	Andy Grove
+    71	Andrew Lamb
+    29	Kun Liu
+    18	kmitchener
+    17	Wei-Ting Kuo
+    14	Yang Jiang
+    13	dependabot[bot]
+    12	Raphael Taylor-Davies
+    11	Batuhan Taskaya
+    11	Kirk Mitchener
```
Comment (andygrove) [inline _posts/2022-10-20-datafusion-13.0.0.md:170]: We should combine stats for `kmitchener` and `Kirk Mitchener`

File: _posts/2022-10-20-datafusion-13.0.0.md (hunk 1 above)
Comment (waitingkuo) [reply _posts/2022-10-20-datafusion-13.0.0.md:48]: it was the result from 12.0. i'll update it soon with the latest version

File: _posts/2022-10-20-datafusion-13.0.0.md (hunk 3 above)
Comment (waitingkuo) [reply _posts/2022-10-20-datafusion-13.0.0.md:114]: - Support TimestampTz ( apache/arrow-datafusion#3660 )

File: _posts/2022-10-20-datafusion-13.0.0.md [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+- A custom Domain Specific Query Language
+- The ability to easily and quickly read and process Parquet, JSON, Avro or CSV data.
+- To read from remote object stores such as AWS S3, Azure Blob Storage, GCP.
+
+Even though DataFusion is 4 years "young," it has seen significant community growth in the last few months and the momentum continues to accelerate.
+
+# Background
+
+
+DataFusion is used as the engine in [many open source and commercial projects](https://github.com/apache/arrow-datafusion#known-uses) and was one of the early open source projects to provide this capability. 2022 has validated our belief in the need for such a ["LLVM for database and AI systems"](https://www.slideshare.net/AndrewLamb32/20220623-apache-arrow-and-datafusion-changing-the-game-for-implementing-database-systemspdf) with announcements such as the [release of FaceBook's Velox](https://engineering.fb.com/2022/08/31/open-source/velox/) engine, the major investments in [Acero](https://arrow.apache.org/docs/cpp/streaming_execution.html) as well as the continued popularity of [Apache Calcite](https://calcite.apache.org/) and other similar technologies.
```
Comment (HaoYang670) [inline _posts/2022-10-20-datafusion-13.0.0.md:44]: I can't access this link: `https://www.slideshare.net/AndrewLamb32/20220623-apache-arrow-and-datafusion-changing-the-game-for-implementing-database-systemspdf`. Could anyone access it?

File: _posts/2022-10-20-datafusion-13.0.0.md [hunk 7]
```diff
@@ -0,0 +1,226 @@
...
+which decouples the two projects and allows each project to move even faster.
+
+Along with numerous other bug fixes and smaller improvements, here are some of the major advances:
+
+## Advanced SQL
+- Support for GROUPING SETS/CUBE/ROLLUP (#2716)
+- Custom window frame logic (support `ROWS`, `RANGE`, `PRECEDING` and `FOLLOWING` for window functions) (#3570)
+- Add support for correlated subqueries (#2885)
+- `ROLLUP` and `CUBE` grouping set expressions (#2446)
+- #2405 SUM DISTINCT aggregate support Sum distinct support
```
Comment (HaoYang670) [inline _posts/2022-10-20-datafusion-13.0.0.md:83]:
```suggestion
- SUM DISTINCT aggregate support Sum distinct support (#2405)
```

Comment (HaoYang670) [reply _posts/2022-10-20-datafusion-13.0.0.md:83]: Also, is it better to append the URLs of the PRs directly?

File: _posts/2022-10-20-datafusion-13.0.0.md (hunk 6 above)
Comment (alamb) [reply _posts/2022-10-20-datafusion-13.0.0.md:44]: https://www.slideshare.net/AndrewLamb32/20220623-apache-arrow-and-datafusion-changing-the-game-for-implementing-database-systemspdf
workf for me:

Screen Shot 2022-10-21 at 3 41 22 PM


However, perhaps the google slides are better; https://docs.google.com/presentation/d/1iNX_35sWUakee2q3zMFPyHE4IV2nC3lkCK_H6Y2qK84/edit#slide=id.p ?

File: _posts/2022-10-20-datafusion-13.0.0.md (hunk 7 above)
Comment (alamb) [reply _posts/2022-10-20-datafusion-13.0.0.md:83]: Yes, you are right -- I will figure out how to get the urls working

File: _posts/2022-10-20-datafusion-13.0.0.md (hunk 2 above)
Comment (alamb) [reply _posts/2022-10-20-datafusion-13.0.0.md:101]: added -- thanks

File: _posts/2022-10-20-datafusion-13.0.0.md (hunk 3 above)
Comment (alamb) [reply _posts/2022-10-20-datafusion-13.0.0.md]: In d02bb30657 👍

File: _posts/2022-10-20-datafusion-13.0.0.md (hunk 4 above)
Comment (alamb) [reply _posts/2022-10-20-datafusion-13.0.0.md:124]: Absolutely -- done in 824648efdf

Comment (alamb) [reply _posts/2022-10-20-datafusion-13.0.0.md:124]: Absolutely -- done in 824648efdf

File: _posts/2022-10-20-datafusion-13.0.0.md (hunk 5 above)
Comment (alamb) [reply _posts/2022-10-20-datafusion-13.0.0.md]: done

File: _posts/2022-10-20-datafusion-13.0.0.md [hunk 8]
```diff
@@ -0,0 +1,284 @@
...
+makes it easier for the increasing number of projects that now depend on DataFusion.
+
+We have also completed the "graduation" of [Ballista to its own top-level arrow-ballista repository](https://docs.google.com/document/d/1jNRbadyStSrV5kifwn0khufAwq6OnzGczG4z8oTQJP4/edit?usp=sharing)
+which decouples the two projects and allows each project to move even faster.
+
+Along with numerous other bug fixes and smaller improvements, here are some of the major advances:
+
+# Improved Support for Cloud Object Stores
+
+DataFusion now has improved support for the major cloud object stores (Amazon S3, Azure Blog Storage, and Google Cloud Storage) via the [object_store](https://crates.io/crates/object_store) crate. Using this integration, DataFusion optimizes reading parquet files by reading only the parts of the files that are needed.
```
Comment (andygrove) [inline _posts/2022-10-20-datafusion-13.0.0.md:80]:
```suggestion
DataFusion now has improved support for the major cloud object stores (Amazon S3, Azure Blob Storage, and Google Cloud Storage) via the [object_store](https://crates.io/crates/object_store) crate. Using this integration, DataFusion optimizes reading parquet files by reading only the parts of the files that are needed.
```

File: _posts/2022-10-20-datafusion-13.0.0.md [hunk 9]
```diff
@@ -0,0 +1,284 @@
...
+
+While Velox and Acero focus on execution engines, DataFusion provides the entire suite of components needed to build most analytic systems, including a SQL frontend, a dataframe API, and  extension points for just about everything. Some [DataFusion users](https://github.com/apache/arrow-datafusion#known-uses) use a subset of the features such as the frontend (e.g. (dask-sql)[https://dask-sql.readthedocs.io/en/latest/] or the execution engine, such as [Blaze](https://github.com/blaze-init/blaze), and some users use many different components to build both SQL based and customized DSL based systems such as [InfluxDB IOx](https://github.com/influxdata/influxdb_iox/pulls) and [VegaFusion](https://github.com/vegafusion/vegafusion).
```
Comment (kou) [inline _posts/2022-10-20-datafusion-13.0.0.md:46]:
```suggestion
While Velox and Acero focus on execution engines, DataFusion provides the entire suite of components needed to build most analytic systems, including a SQL frontend, a dataframe API, and  extension points for just about everything. Some [DataFusion users](https://github.com/apache/arrow-datafusion#known-uses) use a subset of the features such as the frontend (e.g. [dask-sql](https://dask-sql.readthedocs.io/en/latest/) or the execution engine, such as [Blaze](https://github.com/blaze-init/blaze), and some users use many different components to build both SQL based and customized DSL based systems such as [InfluxDB IOx](https://github.com/influxdata/influxdb_iox/pulls) and [VegaFusion](https://github.com/vegafusion/vegafusion).
```

File: _posts/2022-10-20-datafusion-13.0.0.md [hunk 10]
```diff
@@ -0,0 +1,284 @@
...
+
+One of DataFusion’s advantages is its implementation in [Rust](https://www.rust-lang.org/) and thus its easy integration with the broader Rust ecosystem. Rust continues to be a major source of benefit, from the [ease of parallelization with the high quality and standardized `async` ecosystem](https://www.influxdata.com/blog/using-rustlangs-async-tokio-runtime-for-cpu-bound-tasks/) , as well as its modern dependency management system and wonderful performance. <!-- I wonder if we should link to clickbench?? -->
+<!--While we haven’t invested in the benchmarking ratings game datafusion continues to be quite speedy (todo quantity this, with some evidence) – maybe clickbench?-->
+
+<!-- I am not sure this section needed -->
+DataFusion belongs to and builds on the Apache Arrow ecosystem and its family of related technologies and projects in rust. This include
```
Comment (kou) [inline _posts/2022-10-20-datafusion-13.0.0.md:52]:
```suggestion
DataFusion belongs to and builds on the Apache Arrow ecosystem and its family of related technologies and projects in Rust. This include
```

File: _posts/2022-10-20-datafusion-13.0.0.md [hunk 11]
```diff
@@ -0,0 +1,284 @@
...
+ - Reduce `left`/`right`/`full` joins to `inner` join [#2750]
+ - Convert  cross joins to inner joins when possible [#3482]
+ - Sort preserving `SortMergeJoin` [#2699]
+ - Improvements in group by and sort performance [#2375]
+ - Adaptive `regex_replace` implementation [#3518]
+
+# Optimizer Enhancements
+Internally the optimizer has been significantly enhanced as well.
+
+- Casting / coercion now happens logical planning [#3185] [#3396] [#3636]
```
Comment (andygrove) [inline _posts/2022-10-20-datafusion-13.0.0.md:118]:
```suggestion
- Casting / coercion now happens during logical planning [#3185] [#3396] [#3636]
```

File: _posts/2022-10-20-datafusion-13.0.0.md (hunk 6 above)
Comment (alamb) [reply _posts/2022-10-20-datafusion-13.0.0.md:44]: updating to use google slides link

**Comment by alamb** [comment]
I plan to update the dates on this PR and publish it tomorrow unless anyone needs more time to review. Please just let me know if you do so

**Review by andygrove — APPROVED** [review]
LGTM!

File: _posts/2022-10-20-datafusion-13.0.0.md (hunk 6 above)
Comment (js8544) [reply _posts/2022-10-20-datafusion-13.0.0.md]: @alamb The google slides link seems to have protected access. I can't open it with my personal GMail account. The Slideshare link works for me though.

Comment (alamb) [reply _posts/2022-10-20-datafusion-13.0.0.md]: Thanks for the report -- I will add the slideshare link as an alternate one: https://github.com/apache/arrow-site/pull/263

**Pull request merged**